### PR TITLE
[netcpp] new port

### LINKED ISSUE #40252: [New Port Request] netcpp
### Library name

netcpp

### Library description

Simple asynchronous network library

### Source repository URL

https://github.com/index1207/netcpp

### Project homepage (if different from the source repository)

_No response_

### Anything else that is useful to know when adding (such as optional features the library may have that should be included)

_No response_
```[tasklist]
### Tasks
- [ ] https://github.com/microsoft/vcpkg/pull/40245
```

### PR DESCRIPTION
**Pull request by index1207**
fixes #40252

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by index1207** [comment]
@microsoft-github-policy-service agree

**Comment by index1207** [comment]
fixed #40252

**Comment by MonicaLiu0311** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/105681/artifacts?artifactName=failure%20logs%20for%20x64-linux&api-version=7.1&%24format=zip) for `x64-linux`:
```
[1/12] /usr/bin/c++ -DNETCPP_BUILD -I/mnt/vcpkg-ci/b/netcpp/src/v0.3.4-acd1d1d744.clean/src -I/mnt/vcpkg-ci/b/netcpp/src/v0.3.4-acd1d1d744.clean/include -fPIC -g -std=gnu++20 -Winvalid-pch -x c++-header -include /mnt/vcpkg-ci/b/netcpp/x64-linux-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx -MD -MT CMakeFiles/netcpp.dir/cmake_pch.hxx.gch -MF CMakeFiles/netcpp.dir/cmake_pch.hxx.gch.d -o CMakeFiles/netcpp.dir/cmake_pch.hxx.gch -c /mnt/vcpkg-ci/b/netcpp/x64-linux-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx.cxx
FAILED: CMakeFiles/netcpp.dir/cmake_pch.hxx.gch 
/usr/bin/c++ -DNETCPP_BUILD -I/mnt/vcpkg-ci/b/netcpp/src/v0.3.4-acd1d1d744.clean/src -I/mnt/vcpkg-ci/b/netcpp/src/v0.3.4-acd1d1d744.clean/include -fPIC -g -std=gnu++20 -Winvalid-pch -x c++-header -include /mnt/vcpkg-ci/b/netcpp/x64-linux-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx -MD -MT CMakeFiles/netcpp.dir/cmake_pch.hxx.gch -MF CMakeFiles/netcpp.dir/cmake_pch.hxx.gch.d -o CMakeFiles/netcpp.dir/cmake_pch.hxx.gch -c /mnt/vcpkg-ci/b/netcpp/x64-linux-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx.cxx
In file included from /mnt/vcpkg-ci/b/netcpp/x64-linux-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx:5,
                 from <command-line>:
/mnt/vcpkg-ci/b/netcpp/src/v0.3.4-acd1d1d744.clean/include/net/netcpp.hpp:4:10: fatal error: format: No such file or directory
    4 | #include <format>
      |          ^~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/105681/artifacts?artifactName=failure%20logs%20for%20arm64-osx&api-version=7.1&%24format=zip) for `arm64-osx`:
```
[1/12] /Library/Developer/CommandLineTools/usr/bin/c++ -DNETCPP_BUILD -I/Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/src -I/Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/include -fPIC -g -std=gnu++20 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk -Winvalid-pch -Xclang -emit-pch -Xclang -include -Xclang /Users/vcpkg/Data/b/netcpp/arm64-osx-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx -x c++-header -MD -MT CMakeFiles/netcpp.dir/cmake_pch.hxx.pch -MF CMakeFiles/netcpp.dir/cmake_pch.hxx.pch.d -o CMakeFiles/netcpp.dir/cmake_pch.hxx.pch -c /Users/vcpkg/Data/b/netcpp/arm64-osx-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx.cxx
FAILED: CMakeFiles/netcpp.dir/cmake_pch.hxx.pch 
/Library/Developer/CommandLineTools/usr/bin/c++ -DNETCPP_BUILD -I/Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/src -I/Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/include -fPIC -g -std=gnu++20 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk -Winvalid-pch -Xclang -emit-pch -Xclang -include -Xclang /Users/vcpkg/Data/b/netcpp/arm64-osx-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx -x c++-header -MD -MT CMakeFiles/netcpp.dir/cmake_pch.hxx.pch -MF CMakeFiles/netcpp.dir/cmake_pch.hxx.pch.d -o CMakeFiles/netcpp.dir/cmake_pch.hxx.pch -c /Users/vcpkg/Data/b/netcpp/arm64-osx-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx.cxx
In file included from <built-in>:1:
In file included from /Users/vcpkg/Data/b/netcpp/arm64-osx-dbg/CMakeFiles/netcpp.dir/cmake_pch.hxx:5:
In file included from /Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/include/net/netcpp.hpp:7:
/Users/vcpkg/Data/b/netcpp/src/v0.3.4-acd1d1d744.clean/include/net/winsock.hpp:3:10: fatal error: 'WinSock2.h' file not found
#include <WinSock2.h>
         ^~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

File: ports/netcpp/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+netcpp provides CMake targets:
+
+    find_package(netcpp CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE netcpp::netcpp)
```
Comment (MonicaLiu0311) [inline ports/netcpp/usage:4]:
```suggestion
    target_link_libraries(main PRIVATE netcpp::netcpp)

```

Comment (MonicaLiu0311) [reply ports/netcpp/usage:4]: Please leave a blank line.

File: ports/netcpp/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "netcpp",
+  "version": "0.3.4",
+  "description": "A simple asynchronous network library",
+  "homepage": "https://github.com/index1207/netcpp",
```
Comment (index1207) [inline ports/netcpp/vcpkg.json:5]: @MonicaLiu0311 This package supports only windows. Add this line can specify platform?
```json
"supports": "windows"
```

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
netcpp provides CMake targets:

    find_package(netcpp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE netcpp::netcpp)
```

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by MonicaLiu0311** [comment]
If you are ready to be reviewed again, click `Ready for review`.

**Comment by index1207** [comment]
Thank for your review!

File: ports/netcpp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    REPO index1207/netcpp
+    REF "v${VERSION}"
+    SHA512 5f4f99a65ce373f1712bf8fb90e5de4f9945f3467e6535e69fad49898fd8dae1fa534e27070d16557b05ee9fa4e3551900062f6c8220a2bd3b197cce82a57182
+    HEAD_REF release
+)
+
+if(!VCPKG_TARGET_IS_WINDOWS)
+    set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+    return()
+endif()
```
Comment (MonicaLiu0311) [inline ports/netcpp/portfile.cmake:14]:
```suggestion

```

**Review by MonicaLiu0311 — DISMISSED** [review]

File: ports/netcpp/portfile.cmake (hunk 3 above)
Comment (MonicaLiu0311) [reply ports/netcpp/portfile.cmake]: After deleting please run: `./vcpkg x-add-version --overwrite-version netcpp`

**Review by MonicaLiu0311 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**